src/AutoMapper/ConfigurationStore.cs

```diff
@@ -417,10 +417,15 @@ public bool HasOpenGenericTypeMapDefined(ResolutionContext context)
 
         private IEnumerable<TypePair> GetRelatedTypePairs(TypePair root)
         {
-            return
+            var includeOverrideTypePairs =
+                GetAllTypeMaps()
+                    .Where(tm => tm.SourceType.IsAssignableFrom(root.SourceType) && (tm.DestinationTypeOverride ?? tm.DestinationType) != root.DestinationType && tm.DestinationType.IsAssignableFrom(root.DestinationType))
+                    .Select(tm => new TypePair(tm.SourceType,tm.DestinationTypeOverride ?? tm.GetDerivedTypeFor(root.SourceType))).ToList();
```

Comment:
It should hopefully fix #842 as well.  I'll add the example unit test to see if it passes.
I think both are same problem, it's just #842 gets invalid cast exception because it expects inherited type, while other one does not expect it.